python/cudf/cudf/core/column/methods.py

```diff
@@ -11,6 +11,8 @@
 if TYPE_CHECKING:
     from cudf.core.column import ColumnBase
 
+ParentType = Union["cudf.Series", "cudf.Index"]
```

Comment:
Can this become `SingleColumnFrame` from #8115 ?

Reply:
I'm changing this in https://github.com/rapidsai/cudf/pull/8306.